### PR TITLE
[Model Monitoring] Enrich model endpoint with the unique `model_uri` reference 

### PR DESCRIPTION
**Pull request by Eyal-Danieli**
Enrich the model endpoint object with the model uri from the model object with a unique identifier. Instead of using the model uri that includes the model object's name and version, we will use a unique reference that includes the `project`, `key`, `iter`, and `tree` values. This will allow us to handle future changes in the model version without updating the model endpoint object.

Related JIRA: https://iguazio.atlassian.net/browse/ML-5675

### CODE REVIEW

File: tests/system/model_monitoring/test_model_monitoring.py [hunk 1]
```diff
@@ -338,8 +336,9 @@ def test_basic_model_monitoring(self, with_sql_target: bool) -> None:
 
         endpoint = endpoints_list[0]
 
-        # ML-6594
-        assert not endpoint.status.feature_stats
+        if not with_sql_target:
+            # ML-6594
+            assert not endpoint.status.feature_stats
```
Comment (davesh0812) [inline tests/system/model_monitoring/test_model_monitoring.py]: according to https://github.com/mlrun/mlrun/pull/6007 It suppose to be excluded in v3ioKV & SQL no?

Comment (Eyal-Danieli) [reply tests/system/model_monitoring/test_model_monitoring.py]: yes it's still a bug in sql

Comment (davesh0812) [reply tests/system/model_monitoring/test_model_monitoring.py]: OK so as discussed reopen https://iguazio.atlassian.net/browse/ML-6594 or just open a new bug. 
FYI @gtopper

**Review by davesh0812 — APPROVED** [review]
Looks Good, just revert the change in the tests because it already fixed by other pr

**Pull request merged**